core/src/commonMain/kotlin/com/xebia/functional/xef/embeddings/OpenAIEmbeddings.kt

```diff
@@ -15,27 +15,19 @@ class OpenAIEmbeddings(private val config: OpenAIConfig, private val oaiClient:
     texts: List<String>,
     chunkSize: Int?,
     requestConfig: RequestConfig
-  ): List<Embedding> = chunkedEmbedDocuments(texts, chunkSize ?: config.chunkSize, requestConfig)
+  ): List<Embedding> {
+    inline fun toRequest(texts: List<String>): EmbeddingRequest =
+      EmbeddingRequest(requestConfig.model.modelName, texts, requestConfig.user.id)
 
-  override suspend fun embedQuery(text: String, requestConfig: RequestConfig): List<Embedding> =
-    if (text.isNotEmpty()) embedDocuments(listOf(text), null, requestConfig) else emptyList()
+    suspend fun createWithRetry(texts: List<String>): List<Embedding> =
```

Comment:
Nit: This `withRetry` is a bit outdated since that logic is no longer present. It exists directly inside the `HttpClient` now. Perhaps just `createEmbeddings`.

This suggest breaks the code below.

```suggestion
    suspend fun createEmbeddings(texts: List<String>): List<Embedding> =
```